sdk/core/azure-core/inc/azure/core/azure_assert.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Provide assert macros to use as pre-conditions.
+ *
+ * @remark Asserts are turned ON even if `NDEBUG` is defined.
+ *
+ */
+
+#pragma once
+
+#if defined(NDEBUG)
```

Comment:
`NDEBUG`? Why do we use this as the name, rather than just `DEBUG`?

Reply:
NDENUG is validated by assert
https://en.cppreference.com/w/cpp/error/assert

It's the popular and well known symbol for switching from Debug to Prod.  CMake would add this symbol by default when selecting build=Release

Reply:
I see, gotcha. Thanks for the info.